### ISSUE TITLE
notFound handling with route doesn't work 

### ISSUE THREAD

**Issue by pritambarhate** [perwendel/spark#774]
**Spark core version: 2.5.5**

I am trying out a REST API with Spark. I found that though my routes work fine the custom notFound handler is not working. I have basically copy pasted the code from the docs.

```
Spark.notFound((req, res) -> {
			res.type("application/json");
			return "{\"message\":\"Custom 404\"}";
		});
```

When I test this using Postman with a route that is not defined, I get following error:

> Could not get any respose.
> There was an error connecting to: http://localhost:8080/myprj/ssss.

If I hit the same nonexistent URL in Chrome,

> This site can’t be reached
> The webpage at http://localhost:8080/myprj/ssss might be temporarily down or it may have moved permanently to a new web address.

However, at the same time all the routes that are actually defined work fine.

**Comment by tipsy** [comment]
Please provide a minimal code example, seems to work fine here.

**Comment by pritambarhate** [comment]
I have created the minimal code repo [here](https://github.com/pritambarhate/sparkbug/blob/master/src/main/java/sparkbug/Main.java). Also I have found the exact scenario when the above mentioned bug gets reproduced. 

Here is the explanation:

First I tried this as a standalone Java application with a `main` method and it indeed works as expected. However, for my project, I am running spark inside a standard webapp with Tomcat (right now I have Tomcat 8.0.23 installed on my machine.) So I am running Spark via a Filter as explained in the official docs [here](http://sparkjava.com/documentation.html#other-webserver). 

Here I found 2 things. First, in my webapp setup the notFound spark filter has no effect. The request for an URL that doesn't exist returns regular tomcat 404 page. This should not be the case right? It should have given the custom JSON message specified in the notFound handler. Do I need to do anything else in web.xml for the Spark's notFound method to work?

Still, it was not reproducing the behavior I posted in the original comment. So I looked at my project's code again. After adding all my setup code and removing things one by one, I found the exact combination that produces the above results. **When Spark is being used via a Filter in a JEE webapp, along gzip content-encoding filter, the results documented in the original bug report are reproduced.**

The minimal code is repo [here](https://github.com/pritambarhate/sparkbug/blob/master/src/main/java/sparkbug/Main.java) to demonstrate the problem. Running `mvn package` should create the war file which can be tested with Tomcat.

**Comment by tipsy** [comment]
I think the custom error pages are only for the embedded jetty server. I can see if @perwendel wrote anything about it.

**Comment by mlitcher** [comment]
@tipsy, any reason this feature should be limited to the embedded server only?

With a simple change to [MatcherFilter.java](https://github.com/perwendel/spark/blob/master/src/main/java/spark/http/matching/MatcherFilter.java#L161) (remove the externalContainer check), these could be used. I tested deploying to a couple different web containers and it worked fine.

**Comment by tipsy** [comment]
> @tipsy, any reason this feature should be limited to the embedded server only?
Not that I can think of. If you create a PR I can merge it.

**Comment by mlitcher** [comment]
Sure thing. Just submitted #874.

**Comment by tipsy** [comment]
I'll just have a look at the original commit and see if I can figure out why it was done like this.

**Comment by tipsy** [comment]
That `if` was like that before custom error pages were introduced, but I can't figure out why. I can't imagine anyone would prefer the default error pages of the external container?

**Comment by mlitcher** [comment]
I agree. In my particular use case, I'm using an external container and looking to customize my 404 response.

**Comment by pritammobisoft** [comment]
Thanks, @tipsy and @mlitcher for fixing this. Any idea when will this land in a Maven based release?

**Comment by tipsy** [comment]
@pritambarhate next week is the plan.

**Comment by pritammobisoft** [comment]
Thank you.
